R/model_engines.R

```diff
@@ -1,8 +1,96 @@
-# TODO mlp, glm, rulefit, gam, naive bayes,
+# TODO: h2o.gam, h2o.coxph
+class_info <- list(
+  pre = NULL,
+  post = NULL,
+  func = c(pkg = "agua", fun = "h2o_predict_classification"),
+  args = list(
+    object = quote(object$fit),
+    new_data = quote(new_data)
+  )
+)
+class_info_prob <- class_info
+class_info_prob$args$type <- "prob"
+
+reg_info <- list(
+  pre = NULL,
+  post = NULL,
+  func = c(pkg = "agua", fun = "h2o_predict_regression"),
+  args = list(
+    object = quote(object$fit),
+    new_data = quote(new_data)
+  )
+)
+reg_info_raw <- reg_info
+reg_info_raw$args$type <- "raw"
+
+
+add_linear_reg_h2o <- function() {
+  parsnip::set_model_engine("linear_reg", "regression", "h2o")
+  parsnip::set_dependency("linear_reg", "h2o", "h2o")
```

Comment:
Can you use the `mode` argument here (multiple times if there are different modes). The default sets the dependency for all modes. 

This is in a lot of legacy code like this :-/